org.eclipse.lsp4j/src/main/java/org/eclipse/lsp4j/Protocol.xtend

```diff
@@ -3587,6 +3710,108 @@ class RenameParams extends TextDocumentPositionAndWorkDoneProgressParams {
 	}
 }
 
+@Beta
+@JsonRpcData
+class SemanticTokensLegend {
+
+	/**
+	 * The token types that the client supports.
+	 */
+	@NonNull
+	List<String> tokenTypes;
+
+	/**
+	 * The token modifiers that the client supports.
+	 */
+	@NonNull
+	List<String> tokenModifiers;
+
+	new(@NonNull List<String> tokenTypes, @NonNull List<String> tokenModifiers) {
+		this.tokenTypes = Preconditions.checkNotNull(tokenTypes, 'tokenTypes')
+		this.tokenModifiers = Preconditions.checkNotNull(tokenModifiers, 'tokenModifiers')
+	}
+
+}
+
+@Beta
+@JsonRpcData
+class SemanticTokensServerFull {
+
+	/**
+	* The server supports deltas for full documents.
+	*/
+	Boolean delta
+
+	new() {
+	}
+
+	new(Boolean delta) {
+		this.delta = delta
+	}
+}
+
+@Beta
+@JsonRpcData
+class SemanticTokensWithRegistrationOptions extends AbstractWorkDoneProgressOptions {
+
+	/**
+	 * The legend used by the server
+	 */
+	@NonNull
```

Comment:
This property cannot be marked `@NonNull` because it's not present in `SemanticTokensRegistrationOptions`.

Reply:
Actually, it is, according to the [LSP spec](https://microsoft.github.io/language-server-protocol/specifications/specification-3-16/#textDocument_semanticTokens)

Reply:
Yes, but this class represents the TypeScript type `SemanticTokensOptions | SemanticTokensRegistrationOptions`. In that combined type, `legend` is not always present.

Reply:
Because `SemanticTokensRegistrationOptions` is defined as
```typescript
interface SemanticTokensRegistrationOptions extends TextDocumentRegistrationOptions, SemanticTokensOptions, StaticRegistrationOptions {}
```
its set of properties is a superset of `SemanticTokensOptions` meaning it _would_ include `legend`.

Reply:
Oh you are right, I haven't seen the other extended interfaces in the spec because the code line is so long. Then we can leave `@NonNull` as it is.